common/djangoapps/student/views.py

```diff
@@ -1644,6 +1645,12 @@ def create_account_with_params(request, params):
     if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
         params["password"] = pipeline.make_random_password()
 
+    # Add a form requirement for data sharing consent if the EnterpriseCustomer
+    # for the request requires it at login
+    consent_requirement = data_sharing_consent_requirement_at_login(request)
+    if consent_requirement is not None:
+        extra_fields['data_sharing_consent'] = consent_requirement
```

Comment:
Does it make sense to always pass along `data_sharing_consent`, even if the value is None?  Doing so would allow us to simplify this section.